File: dart/difference-of-squares/lib/difference_of_squares.dart
class DifferenceOfSquares {
  
  int squareOfSums(int val) {
    int res = 0;
    for (int i = 1; i <= val; i++) res += i;
    return res * res;
  }
  
  int sumOfSquares(int val) {
    int res = 0;
    for (int i = 1; i <= val; i++) res += i * i;
    return res;
  }
  difference(int val) => squareOfSums(val) - sumOfSquares(val);
}

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares(highLimit: Int) {
    
    private val numbers: List<Int> = 1.rangeTo(highLimit).toList()

    fun sumOfSquares() = numbers.square().sum()
    fun squareOfSum() = numbers.sum().square()
    fun difference() = squareOfSum() - sumOfSquares()

    internal fun Int.square(): Int = this * this
    internal fun List<Int>.square(): List<Int> = this.map { it.square() }
}
